src/main/config/config.c

```diff
@@ -413,291 +428,320 @@ uint16_t getCurrentMinthrottle(void)
-
+    config->blackbox_rate_num = 1;
+    config->blackbox_rate_denom = 1;
 #endif // BLACKBOX
     
     // alternative defaults settings for COLIBRI RACE targets
 #if defined(COLIBRI_RACE)
-    masterConfig.escAndServoConfig.minthrottle = 1025;
-    masterConfig.escAndServoConfig.maxthrottle = 1980;
-    masterConfig.batteryConfig.vbatmaxcellvoltage = 45;
-    masterConfig.batteryConfig.vbatmincellvoltage = 30;
+    config->escAndServoConfig.minthrottle = 1025;
+    config->escAndServoConfig.maxthrottle = 1980;
+    config->batteryConfig.vbatmaxcellvoltage = 45;
+    config->batteryConfig.vbatmincellvoltage = 30;
 #endif
 
-#if defined(TARGET_CONFIG)
-    targetConfiguration(&masterConfig);
-#endif
-    
 #if defined(ALIENFLIGHT) 
-    featureClear(FEATURE_ONESHOT125);
 #ifdef ALIENFLIGHTF1
-    masterConfig.serialConfig.portConfigs[1].functionMask = FUNCTION_RX_SERIAL;
+    config->serialConfig.portConfigs[1].functionMask = FUNCTION_RX_SERIAL;
 #else
-    masterConfig.serialConfig.portConfigs[2].functionMask = FUNCTION_RX_SERIAL;
+    config->serialConfig.portConfigs[2].functionMask = FUNCTION_RX_SERIAL;
 #endif
 #ifdef ALIENFLIGHTF3
-    masterConfig.batteryConfig.vbatscale = 20;
-    masterConfig.mag_hardware = MAG_NONE;            // disabled by default
+    config->batteryConfig.vbatscale = 20;
+    config->mag_hardware = MAG_NONE;            // disabled by default
 #endif
-    masterConfig.rxConfig.serialrx_provider = SERIALRX_SPEKTRUM2048;
-    masterConfig.rxConfig.spektrum_sat_bind = 5;
-    masterConfig.rxConfig.spektrum_sat_bind_autoreset = 1;
-    masterConfig.escAndServoConfig.minthrottle = 1000;
-    masterConfig.escAndServoConfig.maxthrottle = 2000;
-    masterConfig.motor_pwm_rate = 32000;
-    masterConfig.failsafeConfig.failsafe_delay = 2;
-    masterConfig.failsafeConfig.failsafe_off_delay = 0;
-    currentControlRateProfile->rates[FD_PITCH] = 40;
-    currentControlRateProfile->rates[FD_ROLL] = 40;
-    currentControlRateProfile->rates[FD_YAW] = 40;
-    parseRcChannels("TAER1234", &masterConfig.rxConfig);
-
-    masterConfig.customMotorMixer[0] = (motorMixer_t){ 1.0f, -0.414178f,  1.0f, -1.0f };    // REAR_R
-    masterConfig.customMotorMixer[1] = (motorMixer_t){ 1.0f, -0.414178f, -1.0f,  1.0f };    // FRONT_R
-    masterConfig.customMotorMixer[2] = (motorMixer_t){ 1.0f,  0.414178f,  1.0f,  1.0f };    // REAR_L
-    masterConfig.customMotorMixer[3] = (motorMixer_t){ 1.0f,  0.414178f, -1.0f, -1.0f };    // FRONT_L
-    masterConfig.customMotorMixer[4] = (motorMixer_t){ 1.0f, -1.0f, -0.414178f, -1.0f };    // MIDFRONT_R
-    masterConfig.customMotorMixer[5] = (motorMixer_t){ 1.0f,  1.0f, -0.414178f,  1.0f };    // MIDFRONT_L
-    masterConfig.customMotorMixer[6] = (motorMixer_t){ 1.0f, -1.0f,  0.414178f,  1.0f };    // MIDREAR_R
-    masterConfig.customMotorMixer[7] = (motorMixer_t){ 1.0f,  1.0f,  0.414178f, -1.0f };    // MIDREAR_L#endif
+    config->rxConfig.serialrx_provider = SERIALRX_SPEKTRUM2048;
```

Comment:
`src/main/config/config.c` is quite messy indeed. The default config seems to have a lot of repeated assignments to the same field, which probably got added when somebody appended new defaults instead of adjusting the existing ones.

I would be in to try and clean up `src/main/config/config.c` in a future chunk of work, potentially alongside with splitting the generic values in master_t into a type of their own, and moving this out of `src/main/config/config_master.h`, so that they can be accessed without having to include 20 or so header files for everything that is referenced in `src/main/config/config_master.h`.